### ISSUE TITLE
How to reconfigure authentication from deprecated user/password to token?

### ISSUE THREAD

**Issue by adswa** [datalad/datalad#5987]
I saw this in the context of trying out @mih's PR #5949 and its probably a really stupid question.
My setup is as follows: I have a configured for GitHub user and password in my keyring.
I also have a configured Github token (I left a comment in #5949 that this token is identified as "datalad-None" in my keyring). It works fine, it queries interactively for the token, and succeeds in creating a GitHub sibling.

When I push to this sibling, ``push`` seems to fall back to user password authentication. This fails due to deprecation:

```
(handbook2) adina@muninn in /tmp/ds on git:master
❱ datalad create-sibling-github new-create-sibling-command-gh          
create_sibling_github(ok): [sibling repository 'github' created at https://github.com/adswa/new-create-sibling-command-gh]
configure-sibling(ok): . (sibling)
action summary:
  configure-sibling (ok: 1)
  create_sibling_github (ok: 1)
(handbook2) adina@muninn in /tmp/ds on git:master
❱ datalad push --to github
Update availability for 'github':  75%|▊| 3.00/4.00 [00:00<00:00, 14.7k Steps/s]Username for 'https://github.com': adswa
Password for 'https://adswa@github.com': <mypw>
CommandError: 'git -c diff.ignoreSubmodules=none push --progress --porcelain github master:master' failed with exitcode 128 under /tmp/ds
remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
fatal: Authentication failed for 'https://github.com/adswa/new-create-sibling-command-gh.git/'
```

Counter-intuitively for most people, it would work when instead of supplying user name and password the token is given:

```
(handbook2) adina@muninn in /tmp/ds on git:master
❱ datalad push --to github                                          
Update availability for 'github':  75%|▊| 3.00/4.00 [00:00<00:00, 31.1k Steps/s]
Username for 'https://github.com': <mytoken>
Password for 'https://<mytoken>@github.com': <mytoken>                                                                               
Username for 'https://github.com': <mytoken>      
Password for 'https://<mytoken>@github.com': 
publish(ok): . (dataset) [refs/heads/master->github:refs/heads/master [new branch]]                                                                             
publish(ok): . (dataset) [refs/heads/git-annex->github:refs/heads/git-annex [new branch]]
action summary:                                                                 
  publish (ok: 2)
```
But how can I get it to prompt for token authentication?
If I remove all github related passwords and user names from my keyring, it still tries to go with user password.

**Comment by yarikoptic** [comment]
- I guess it is maint/master version (not #5949, which I think removes no longer supported user/password authentication)?
  - then I guess this issue would self-resolve whenever that one is merged? but that would probably be for 0.16.0 :-/  so we might better just drop user/password auth attempt for this command altogether for 0.15.x
- in the last example `<mytoken>` should be removed from username/password entry, right?  i.e. you didn't supply them

**Comment by adswa** [comment]
I think I realized what was the problem. I talked to an https URL, which expects user name and password (and accepts a token in place of a password, but wouldn't prompt for "token" explicitly). If I were to change the format of this url to something like
``git remote add origin https://<TOKEN>@github.com/<USERNAME>/<REPO>.git`` it should work
